cmd/performance-profile-creator/cmd/root.go

```diff
@@ -139,6 +139,10 @@ func NewRootCommand() *cobra.Command {
 		Use:   "performance-profile-creator",
 		Short: "A tool that automates creation of Performance Profiles",
 		RunE: func(cmd *cobra.Command, args []string) error {
+			mustGatherDirPath := cmd.Flag("must-gather-dir-path").Value.String()
+			if err := validateMustGatherDirPath(mustGatherDirPath); err != nil {
+				return err
+			}
```

Comment:
well intentioned, but nothing guarantees that the path will be valid after this check: https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use

for maximum safety we should do the check as late as possible (so there's minimally, ideally none, difference between time to check and time to use)

Reply:
I don't know at what stage of the PR you have reviewed it but if you'll have a look at the latest revision you'll see the path is being used immediately after the validation.